horovod/runner/common/util/hosts.py

```diff
@@ -28,29 +30,21 @@ def from_string(host_string):
         return HostInfo(hostname, int(slots))
 
 
+@dataclass
 class SlotInfo:
-    def __init__(self, hostname, rank, local_rank, cross_rank, size=None, local_size=None, cross_size=None):
-        self.hostname = hostname
-        self.rank = rank
-        self.size = size
-        self.local_rank = local_rank
-        self.local_size = local_size
-        self.cross_rank = cross_rank
-        self.cross_size = cross_size
+    hostname: str
+    rank: int
+    local_rank: int
+    cross_rank: int
+    size: int
+    local_size: int
```

Comment:
`Optional[int]` makes the optional parameters nullable, maybe that is equivalent to earlier `__init__`...

Reply:
The new initialization has all fields defined at initialization, so no more nullable needed :)